### PR TITLE
Update library, SBT plugin, SBT, and Scala versions

### PR DESCRIPTION
**Pull request by SeanCheatham**
## Purpose
- Updates several dependencies to the latest versions
## Approach
- Used IntelliJ and https://index.scala-lang.org/
## Testing
- GitHub Actions :shrug: 
## Tickets
N/A

### CODE REVIEW

File: typeclasses/src/main/scala/co/topl/typeclasses/Prepend.scala [hunk 1]
```diff
@@ -1,17 +1,18 @@
 package co.topl.typeclasses
 
 import cats.data.{NonEmptyChain, NonEmptyVector}
-import simulacrum.typeclass
 
 /**
  * Typeclass indicating that something can prepend elements
  */
-@typeclass trait Prepend[F[_]] {
```
Comment (SeanCheatham) [inline typeclasses/src/main/scala/co/topl/typeclasses/Prepend.scala:9]: Issues with `@typeclass` + `F[_]` + Scala 2.13.8

File: project/plugins.sbt [hunk 2]
```diff
@@ -4,19 +4,23 @@ logLevel := Level.Error
...
+  "ch.epfl.scala"           % "sbt-scalafix"              % "0.10.1",
+  "org.wartremover"         % "sbt-wartremover"           % "3.0.5",
+  "com.github.sbt"          % "sbt-native-packager"       % "1.9.10",
+  "com.eed3si9n"            % "sbt-buildinfo"             % "0.11.0",
+  "com.github.sbt"          % "sbt-ci-release"            % "1.5.10",
+  "net.bzzt"                % "sbt-reproducible-builds"   % "0.30",
+  "com.lightbend.akka.grpc" % "sbt-akka-grpc"             % "2.1.5"
 ).map(addSbtPlugin)
+
+ThisBuild / libraryDependencySchemes ++= Seq(
```
Comment (SeanCheatham) [inline project/plugins.sbt:24]: scala-xml dependency changes between SBT plugin versions, and this was one way around it

File: typeclasses/src/main/scala/co/topl/typeclasses/NonEmpty.scala [hunk 3]
```diff
@@ -1,17 +1,18 @@
 package co.topl.typeclasses
 
 import cats.data.{NonEmptyChain, NonEmptyVector}
-import simulacrum.typeclass
 
 /**
  * Typeclass indicating something is non-empty
  */
-@typeclass trait NonEmpty[F[_]] {
```
Comment (scasplte2) [inline typeclasses/src/main/scala/co/topl/typeclasses/NonEmpty.scala:9]: 🤦 it was this easy? I'd been hitting this issue with the updated Scala version for a few months now. 👍 appreciate it being resolved

**Review by scasplte2 — APPROVED** [review]
⬆️

File: typeclasses/src/main/scala/co/topl/typeclasses/NonEmpty.scala (hunk 3 above)
Comment (SeanCheatham) [reply typeclasses/src/main/scala/co/topl/typeclasses/NonEmpty.scala:9]: It's not quite resolved :frowning: 

There's an open [PR](https://github.com/typelevel/simulacrum/pull/226) in simulacrum which hits the same issue.  But this workaround at least allows us to upgrade.

File: project/plugins.sbt (hunk 2 above)
Comment (SeanCheatham) [reply project/plugins.sbt:25]: Yup fair point, updated with a link to the SBT Issue

**Pull request merged**